widget/select.go

```diff
@@ -107,20 +110,67 @@ type Select struct {
 	PlaceHolder string
 	OnChanged   func(string) `json:"-"`
 
-	hovered, tapped bool
-	popUp           *PopUpMenu
+	hovered bool
+	focused bool
+	tapped  bool
+	popUp   *PopUpMenu
+	Parent  fyne.CanvasObject
 }
 
 var _ fyne.Widget = (*Select)(nil)
 
+// FocusGained is called when the Entry has been given focus.
+func (s *Select) FocusGained() {
+	s.focused = true
+	s.Refresh()
+}
+
+// FocusLost is called when the Entry has had focus removed.
+func (s *Select) FocusLost() {
+	s.focused = false
+	s.Refresh()
+}
+
+// Focused returns whether or not this Entry has focus.
+func (s *Select) Focused() bool {
+	return s.focused
+}
+
+func (s *Select) TypedRune(r rune) {
+}
+
+func (s *Select) TypedKey(key *fyne.KeyEvent) {
+	if key.Name == fyne.KeyEnter || key.Name == fyne.KeyReturn || key.Name == fyne.KeySpace {
+		if s.popUp==nil || !s.popUp.Visible() {
+			s.Tapped(nil)
+		} else {
+			s.popUp.Menu.Do()
+		}
+	}
+	if s.popUp!=nil {
+		if key.Name == fyne.KeyUp {
+			s.popUp.Menu.Up()
+		} else if key.Name == fyne.KeyDown {
+			s.popUp.Menu.Down()
+		} else if key.Name == fyne.KeyEscape {
+			s.popUp.Dismiss()
+		}
+	}
+}
+
+func (s *Select) KeyUp(key *fyne.KeyEvent) {
```

Comment:
Why adding no-ops?

Reply:
Well, I do not know why I added them. Now I have removed them.